kratos/python_scripts/auxiliar_methods_adaptative_remeshing.py

```diff
@@ -0,0 +1,151 @@
+            {
+                "list_model_parts_to_assign_conditions" : []
+            }
+            """)
+            root_model_part_name = computing_model_part.GetRootModelPart().Name
+            for name in list_model_parts:
+                name = name.replace(root_model_part_name + ".", "")
+                skin_detection_parameters["list_model_parts_to_assign_conditions"].Append(name)
+
+            if computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess2D(computing_model_part, skin_detection_parameters)
+            else:
+                detect_skin = KratosMultiphysics.SkinDetectionProcess3D(computing_model_part, skin_detection_parameters)
+            detect_skin.Execute()
+        solver.SetEchoLevel(self.analysis.echo_level)
+
+    def AdaptativeRemeshingRunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage for cases where remeshing may be considered
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+
+        # If we remesh using a process
+        computing_model_part = self.analysis._GetSolver().GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+
+        while self.analysis.KeepAdvancingSolutionLoop():
+            self.analysis.time = self.analysis._GetSolver().AdvanceInTime(self.analysis.time)
+            # We reinitialize if remeshed previously
+            if root_model_part.Is(KratosMultiphysics.MODIFIED):
+                self.analysis.ReInitializeSolver()
+            self.analysis.InitializeSolutionStep()
+            # We reinitialize if remeshed on the InitializeSolutionStep
+            if root_model_part.Is(KratosMultiphysics.MODIFIED):
+                self.analysis.ReInitializeSolver()
+                self.analysis.InitializeSolutionStep()
+            self.analysis._GetSolver().Predict()
+            self.analysis._GetSolver().SolveSolutionStep()
+            self.analysis.FinalizeSolutionStep()
+            self.analysis.OutputSolutionStep()
+
+    def SPRAdaptativeRemeshingRunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage for cases where remeshing may be considered with SPR convergence criteria
+
+            Keyword arguments:
+            self It signifies an instance of a class.
+        """
+
+        # Remeshing adaptively
+        solver = self.analysis._GetSolver()
+        computing_model_part = solver.GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+        convergence_criteria = solver.get_convergence_criterion()
+        builder_and_solver = solver.get_builder_and_solver()
+        mechanical_solution_strategy = solver.get_mechanical_solution_strategy()
+
+        while self.analysis.KeepAdvancingSolutionLoop():
+            self.analysis.time = solver.AdvanceInTime(self.analysis.time)
```

Comment:
This is a full-fledged analysis stage... Could you not include the outermost functions in your own analysis stage and break down the most reusable parts into processes for everyone to use? (from @KratosMultiphysics/implementation-committee )

Reply:
I do not understand